### ISSUE TITLE
[Workflow Feature] Deferred Cancelling

### ISSUE THREAD

**Issue by geomagilles** [infiniticio/infinitic#60]
Currently, from a workflow we can wait completion or check the status of a deferred - it could be useful to be able to cancel a Deferred (task or workflow)

**Comment by ManokarG** [comment]
Looking forward for release of this feature.

**Comment by geomagilles** [comment]
@ManokarG can you please elaborate about your use cases?

**Comment by ManokarG** [comment]
Yes, @geomagilles I have to call some third party API and wait for the response and also wants to set a timer for 10 mins if the timer hits first i want to cancel that task so that means i have to call a function in that task to cancel the deferred.

**Comment by geomagilles** [comment]
> i want to cancel that task so that means i have to call a function in that task to cancel the deferred.

You would like to be able to call a function directly in the task implementation to cancel it?

**Comment by ManokarG** [comment]
Yes that's correct.

**Comment by geomagilles** [comment]
At short term, the cancellation of a deferred will have the following consequence:
- trying to await() a canceled deferred will throw an exception
- if the deferred is a child workflow, this workflow will be stopped

Adding a method directly in the task executors to handle gracefully the cancellation (if it's currently running) is possible  but more difficult and could be done later
